### ISSUE TITLE
Album art is not always accurate

### ISSUE THREAD

**Issue by mattgeowild** [musa11971/Music-Bar#31]
### The album shown when clicking on the music bar doesn't always match the album that it is really from, as shown in the music app.
For example, listening to "Dirt (Acoustic),"  from the album _Acoustic Sessions_ by Florida Georgia Line, shows as "Dirt (Acoustic)," from the album _Dirt (Acoustic Remix) - Single_.

**Comment by musa11971** [comment]
This is due to the fact that album covers are retrieved through the iTunes API, which is not always accurate. 

If anyone has ideas on how to improve this, I'd love to hear it.

**Comment by timlance** [comment]
In my case album art is embedded in each file itself. I have many many tracks that are from live recordings not commercially acquired (Dimeadozen, LosslessLegs, etc.). The art is something I made and then embedded when I converted files from FLAC to ALAC. Other apps like MusicBar use this art. If MusicBar could do it MusicBar would be my app of choice. (Currently it is TunesArt)

**Comment by ApaeP** [comment]
I noticed that for most of my french tracks, they are inaccurate.
It would be nice to have an option to hide the artwork then and have something more or less like a miniplayer with a dark background.
Otherwise the app works wonderfully and it's a pleasure to use it, thank you

**Comment by onodera-punpun** [comment]
>edit: I know there are ways of directly pulling the album art via AppleScript, but I haven't been able to get that to work.

Have you ever figured something out regarding this?

**Comment by musa11971** [comment]
> > edit: I know there are ways of directly pulling the album art via AppleScript, but I haven't been able to get that to work.
> 
> Have you ever figured something out regarding this?

Unfortunately not. Music Bar still uses the iTunes API to find artwork.

**Comment by onodera-punpun** [comment]
I spend some time researching this yesterday, and the way updating the get artwork applescript to:

```
        tell application "Music"
            if it is running then
                get raw data of artwork 1 of current track
            end if
        end tell
```

Then put the output into `NSImage(data: <here>)`.

**Comment by musa11971** [comment]
Implemented your solution in v1.5 @onodera-punpun. Thanks 🎉
